Assets/MixedRealityToolkit.Tests/PlayModeTests/InputSystem/GazeProviderTests.cs

```diff
@@ -0,0 +1,108 @@
+            Assert.IsTrue(CoreServices.InputSystem.GazeProvider.GazeCursor.IsVisible);
+
+            // Showing the hand should cause the gaze cursor to be not visible.
+            var inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+            yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSimulationService);
+
+            // Verify that since the hand is shown, the gaze cursor is not visible.
+            Assert.IsFalse(CoreServices.InputSystem.GazeProvider.GazeCursor.IsVisible);
+        }
+
+        /// <summary>
+        /// Verifies that when the gaze cursor visibility override is set to false,
+        /// the gaze cursor is hidden (regardless of the current state) and that
+        /// setting it back to null will undo the hidden transition.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestFalseGazeCursorVisibilityOverride()
+        {
+            // The gaze cursor should be initially visible
+            Assert.IsTrue(CoreServices.InputSystem.GazeProvider.GazeCursor.IsVisible);
+
+            // Set the override to false, and then wait a frame in order for the
+            // internal state machine to reconcile this property change.
+            CoreServices.InputSystem.GazeProvider.GazeCursorVisibilityOverride = false;
+            yield return null;
+            Assert.IsFalse(CoreServices.InputSystem.GazeProvider.GazeCursor.IsVisible);
+
+            // Clear the override and validate that the cursor shows up again.
+            CoreServices.InputSystem.GazeProvider.GazeCursorVisibilityOverride = null;
+            yield return null;
+            Assert.IsTrue(CoreServices.InputSystem.GazeProvider.GazeCursor.IsVisible);
+        }
+
+        /// <summary>
+        /// Verifies that when the gaze cursor visibility override is set to true,
+        /// the gaze cursor is visible (regardless of the current state) and that
+        /// setting it back to null will undo the visible transition.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestTrueGazeCursorVisibilityOverride()
+        {
+            // The gaze cursor should be initially visible
+            Assert.IsTrue(CoreServices.InputSystem.GazeProvider.GazeCursor.IsVisible);
+
+            // Showing the hand should cause the gaze cursor to be not visible.
+            var inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+            yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSimulationService);
+
+            // Verify that since the hand is shown, the gaze cursor is not visible.
+            Assert.IsFalse(CoreServices.InputSystem.GazeProvider.GazeCursor.IsVisible);
+
+            // Set the override to true, and then wait a frame in order for the
+            // internal state machine to reconcile this property change.
+            CoreServices.InputSystem.GazeProvider.GazeCursorVisibilityOverride = false;
```

Comment:
it says in the comment that the override is set to true, but you're setting it to false?

Reply:
also shouldn't the asset fail after setting it to false because the gaze cursor should be hidden?